kornia/nerf/nerf_solver.py

```diff
@@ -22,25 +22,24 @@ class NerfSolver:
         device: device for class tensors: Union[str, torch.device]
         dtype: type for all floating point calculations: torch.dtype
     """
+    _nerf_model: Module
+    _opt_nerf: optim.Optimizer
 
-    def __init__(self, device: Device, dtype: torch.dtype) -> None:
+    def __init__(self, device: torch.device, dtype: torch.dtype) -> None:
         self._cameras: Optional[PinholeCamera] = None
         self._min_depth: float = 0.0
         self._max_depth: float = 0.0
         self._ndc: bool = True
 
         self._imgs: Optional[Images] = None
-        self._num_img_rays: Optional[int] = None
+        self._num_img_rays: Optional[Union[Tensor, int]] = None
 
         self._batch_size: int = 0
 
-        self._nerf_model: Optional[nn.Module] = None
         self._num_ray_points: int = 0
 
         self._nerf_optimizaer: Optional[optim.Optimizer] = None
 
-        self._opt_nerf: optim.Optimizer = None
```

Comment:
Removing `self._opt_nerf` here could be problematic. It is assigned in `init_training`, and if this function is not called, it is undefined! That's why I originally initialized it to `None` here

Reply:
@YanivHollander maybe this wasn't tested ? we didn't see any fail during the PR